Symfony/CS/Tokenizer/Token.php

```diff
@@ -99,9 +100,22 @@ public function equals($other)
         $selfPrototype = $this->getPrototype();
 
         foreach ($otherPrototype as $key => $val) {
-            if ($selfPrototype[$key] !== $val) {
+            // Make sure the token has such key
+            if (!isset($selfPrototype[$key])) {
                 return false;
             }
+
+            if (1 === $key && !$caseSensitive) {
+                // Case-insensitive comparison only applies to the content (key 1)
+                if (0 !== strcasecmp($val, $selfPrototype[1])) {
```

Comment:
In your tests `strcasecmp($val, $selfPrototype[1])` is always `0`.
Please add more test cases to cover this properly.